### PR TITLE
[templates/nextjs][BYOC] Fast follower for client BYOC support

### PR DESCRIPTION
**Pull request by art-alexeyenko**
- [x] This PR follows the [Contribution Guide](https://github.com/Sitecore/jss/blob/dev/CONTRIBUTING.md)
- [x] Changelog updated

## Description / Motivation
Some issues were found with client-side BYOC components and the SitecoreForm import. This PR updates feaas-clientside dependency which addresses them:
1) Flickering of the error frame component when client component is loading (i.e. SitecoreForm)
2) Page occasionally crashing when adding a BYOC component in Pages

## Testing Details
- [x] Unit Test Adjusted
- [ ] Manual Test/Other (Please elaborate)

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/create-sitecore-jss/src/templates/nextjs/src/byoc/index.client.tsx [hunk 1]
```diff
@@ -8,6 +8,8 @@ import * as FEAAS from '@sitecore-feaas/clientside/react';
 // SitecoreForm component which enables new Sitecore Forms for XMCloud functionality. Requires a license to be available in Pages.
 // Sitecore Forms for Sitecore XP are still available separately via @sitecore-jss-forms package
 import '@sitecore/components/form';
+import React from 'react';
+import { ComponentProps } from 'react';
```
Comment (ambrauer) [inline packages/create-sitecore-jss/src/templates/nextjs/src/byoc/index.client.tsx:12]: Should we move these to the top to disassociate them from the component imports?

File: packages/create-sitecore-jss/src/templates/nextjs/src/byoc/index.client.tsx [hunk 2]
```diff
@@ -8,6 +8,8 @@ import * as FEAAS from '@sitecore-feaas/clientside/react';
 // SitecoreForm component which enables new Sitecore Forms for XMCloud functionality. Requires a license to be available in Pages.
```
Comment (ambrauer) [inline packages/create-sitecore-jss/src/templates/nextjs/src/byoc/index.client.tsx:8]: minor language suggestions
```suggestion
// Sitecore Form component which enables Forms for XM Cloud functionality. Requires a license to be available.
```

File: packages/create-sitecore-jss/src/templates/nextjs/src/byoc/index.client.tsx [hunk 3]
```diff
@@ -16,4 +18,6 @@ import '@sitecore/components/form';
  */
 
 // An important boilerplate component that prevents BYOC components from being optimized away. Should be kept in this file.
-export default (props: unknown) => FEAAS.ExternalComponent(props);
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const DefaultExport = (props: ComponentProps) => FEAAS.ExternalComponent(props) || <></>;
```
Comment (ambrauer) [inline packages/create-sitecore-jss/src/templates/nextjs/src/byoc/index.client.tsx:22]: Curious, why the need to fallback to `<></>` here? In what scenario would `FEAAS.ExternalComponent` return falsy value?

**Review by ambrauer — COMMENTED** [review]
Looks good, just a few questions/suggestions

File: packages/create-sitecore-jss/src/templates/nextjs/src/byoc/index.client.tsx (hunk 1 above)
Comment (art-alexeyenko) [reply packages/create-sitecore-jss/src/templates/nextjs/src/byoc/index.client.tsx:12]: yes, good idea

File: packages/create-sitecore-jss/src/templates/nextjs/src/byoc/index.client.tsx (hunk 2 above)
Comment (art-alexeyenko) [reply packages/create-sitecore-jss/src/templates/nextjs/src/byoc/index.client.tsx:8]: SitecoreForm is the component name - so it's intended to call it so here, but the description can be simplified overall

File: packages/create-sitecore-jss/src/templates/nextjs/src/byoc/index.client.tsx (hunk 3 above)
Comment (art-alexeyenko) [reply packages/create-sitecore-jss/src/templates/nextjs/src/byoc/index.client.tsx:22]: FEAAS.ExternalComponent return type has been changed from Component to Component  | undefined. This causes typescript errors in index.ts when bundle is loaded with dynamic() call, and this is a hack to fix that.

Comment (ambrauer) [reply packages/create-sitecore-jss/src/templates/nextjs/src/byoc/index.client.tsx:22]: Got it. Maybe a question for @Inviz then

File: CHANGELOG.md [hunk 4]
```diff
@@ -13,10 +13,9 @@ Our versioning strategy is as follows:
 
 ### 🎉 New Features & Improvements
 
-* `[templates/nextjs]` Enable client-only BYOC component imports. Client-only components can be imported through src/byoc/index.client.ts. Hybrid (server render with client hydration) components can be imported through src/byoc/index.hybrid.ts. BYOC scaffold logic is also moved from nextjs-sxa addon into base template ([#1628](https://github.com/Sitecore/jss/pull/1628))
+* `[templates/nextjs]` Enable client-only BYOC component imports. Client-only components can be imported through src/byoc/index.client.ts. Hybrid (server render with client hydration) components can be imported through src/byoc/index.hybrid.ts. BYOC scaffold logic is also moved from nextjs-sxa addon into base template ([#1628](https://github.com/Sitecore/jss/pull/1628)[#1636](https://github.com/Sitecore/jss/pull/1636))
 
 * `[templates/nextjs]` Import SitecoreForm component into sample nextjs app ([#1628](https://github.com/Sitecore/jss/pull/1628))
-* `[sitecore-jss-nextjs]` (Vercel/Sitecore) Deployment Protection Bypass support for editing integration. ([#1634](https://github.com/Sitecore/jss/pull/1634))
```
Comment (ambrauer) [inline CHANGELOG.md:18]: Looks like this was inadvertently removed during merge conflict resolution

**Review by ambrauer — APPROVED** [review]
Just the one fix for CHANGELOG, otherwise looks good!

File: CHANGELOG.md (hunk 4 above)
Comment (art-alexeyenko) [reply CHANGELOG.md:18]: fixed

**Pull request merged**